File: libs/shared-auth/src/main/java/br/com/deltaglobalbank/sharedauth/JwtAuthenticationToken.java
package br.com.deltaglobalbank.sharedauth;

import java.util.List;

import org.springframework.security.authentication.AbstractAuthenticationToken;
import org.springframework.security.core.authority.SimpleGrantedAuthority;

public class JwtAuthenticationToken extends AbstractAuthenticationToken {

    private final AuthenticatedPrincipal authenticatedPrincipal;

    public JwtAuthenticationToken(AuthenticatedPrincipal authenticatedPrincipal) {
        super(authenticatedPrincipal.roles().stream()
            .map(role -> new SimpleGrantedAuthority("ROLE_" + role))
            .map(authority -> (org.springframework.security.core.GrantedAuthority) authority)
            .toList());
        this.authenticatedPrincipal = authenticatedPrincipal;
        setAuthenticated(true);
    }

    @Override
    public Object getCredentials() {
        return null;
    }

    @Override
    public AuthenticatedPrincipal getPrincipal() {
        return authenticatedPrincipal;
    }
}

File: libs/shared-auth/src/main/kotlin/br/com/deltaglobalbank/sharedauth/JwtAuthenticationToken.kt
package br.com.deltaglobalbank.sharedauth

import org.springframework.security.authentication.AbstractAuthenticationToken
import org.springframework.security.core.authority.SimpleGrantedAuthority

class JwtAuthenticationToken(
    private val authenticatedPrincipal: AuthenticatedPrincipal
) : AbstractAuthenticationToken(
    authenticatedPrincipal.roles.map { SimpleGrantedAuthority("ROLE_$it") }
) {
    init {
        isAuthenticated = true
    }

    override fun getCredentials(): Any? = null
    override fun getPrincipal(): AuthenticatedPrincipal = authenticatedPrincipal
}
